Composer/plugins/runtimes/src/index.ts

```diff
@@ -0,0 +1,186 @@
+      }
+
+      // do the dotnet publish
+      try {
+        const { stdout, stderr } = await exec(
+          `dotnet publish "${dotnetProjectPath}" -c release -o "${publishFolder}" -v q`,
+          {
+            cwd: runtimePath,
+          }
+        );
+        composer.log('OUTPUT FROM BUILD', stdout);
+        if (stderr) {
+          console.error('ERR FROM BUILD: ', stderr);
+        }
+      } catch (err) {
+        console.error('Error doing dotnet publish', err);
+        throw err;
+        return;
+      }
+      // Then, copy the declarative assets into the build artifacts folder.
+      const remoteBotPath = path.join(publishFolder, 'ComposerDialogs');
+      const localBotPath = path.join(runtimePath, 'ComposerDialogs');
+      await fs.copy(localBotPath, remoteBotPath, {
+        overwrite: true,
+        recursive: true,
+      });
+
+      // write settings to disk in the appropriate location
+      const settingsPath = path.join(publishFolder, 'ComposerDialogs', 'settings', 'appsettings.json');
+      // Set the bot and root fields to `ComposerDialogs` - this points the runtime to the appropriate deployed location
+      // todo: are both necessary?
+      Object.assign(settings, { bot: 'ComposerDialogs', root: 'ComposerDialogs' });
+      if (!(await fs.pathExists(path.dirname(settingsPath)))) {
+        fs.mkdirSync(path.dirname(settingsPath), { recursive: true });
+      }
+      fs.writeFileSync(settingsPath, JSON.stringify(settings, null, 2));
+
+      // return the location of the build artifiacts
+      return publishFolder;
+    },
+    eject: async (project, localDisk: IFileStorage) => {
+      const sourcePath = path.resolve(__dirname, '../../../../runtime/dotnet');
+      const destPath = path.join(project.dir, 'runtime');
+      if (!(await project.fileStorage.exists(destPath))) {
+        // used to read bot project template from source (bundled in plugin)
+        await copyDir(sourcePath, localDisk, destPath, project.fileStorage);
+        const schemaDstPath = path.join(project.dir, 'schemas');
+        const schemaSrcPath = path.join(sourcePath, 'azurewebapp/schemas');
+        const customSchemaExists = fs.existsSync(schemaDstPath);
+        const pathsToExclude: Set<string> = new Set();
+        if (customSchemaExists) {
+          const sdkExcludePath = await localDisk.glob('sdk.schema', schemaSrcPath);
+          if (sdkExcludePath.length > 0) {
+            pathsToExclude.add(path.join(schemaSrcPath, sdkExcludePath[0]));
+          }
+        }
+        await copyDir(schemaSrcPath, localDisk, schemaDstPath, project.fileStorage, pathsToExclude);
+        const schemaFolderInRuntime = path.join(destPath, 'azurewebapp/schemas');
+        await removeDirAndFiles(schemaFolderInRuntime);
+        return destPath;
+      }
+      throw new Error(`Runtime already exists at ${destPath}`);
+    },
+  });
+
+  composer.addRuntimeTemplate({
+    key: 'javascript-azurewebapp',
```

Comment:
in the directory structure I think we agreed on calling this `node` instead of `javascript`. Should we make this `node-azurewebapp` or `nodejs-azurewebapp`? I'm flexible here but ideally it should be the same as the directory name under `runtime` for the js runtime.

Reply:
makes sense to me.